diff -Naur squid-3.5.11/src/tools.cc squid-3.5.11-new/src/tools.cc
--- squid-3.5.11/src/tools.cc	2015-11-01 11:44:25.000000000 +0100
+++ squid-3.5.11-new/src/tools.cc	2015-11-08 22:45:01.157330000 +0100
@@ -873,7 +873,7 @@
 void
 setSystemLimits(void)
 {
-#if HAVE_SETRLIMIT && defined(RLIMIT_NOFILE) && !_SQUID_CYGWIN_
+#if HAVE_SETRLIMIT && defined(RLIMIT_NOFILE)
     /* limit system filedescriptors to our own limit */
 
     /* On Linux with 64-bit file support the sys/resource.h header
@@ -896,7 +896,7 @@
     }
 #endif /* HAVE_SETRLIMIT */
 
-#if HAVE_SETRLIMIT && defined(RLIMIT_DATA) && !_SQUID_CYGWIN_
+#if HAVE_SETRLIMIT && defined(RLIMIT_DATA)
     if (getrlimit(RLIMIT_DATA, &rl) < 0) {
         debugs(50, DBG_CRITICAL, "getrlimit: RLIMIT_DATA: " << xstrerror());
     } else if (rl.rlim_max > rl.rlim_cur) {
@@ -912,7 +912,7 @@
         debugs(50, DBG_IMPORTANT, "NOTICE: Could not increase the number of filedescriptors");
     }
 
-#if HAVE_SETRLIMIT && defined(RLIMIT_VMEM) && !_SQUID_CYGWIN_
+#if HAVE_SETRLIMIT && defined(RLIMIT_VMEM)
     if (getrlimit(RLIMIT_VMEM, &rl) < 0) {
         debugs(50, DBG_CRITICAL, "getrlimit: RLIMIT_VMEM: " << xstrerror());
     } else if (rl.rlim_max > rl.rlim_cur) {
